stl/inc/random

```diff
@@ -1116,27 +1118,45 @@ protected:
     _Post_satisfies_(this->_Idx == 0)
 
         void _Refill_lower() { // compute values for the lower half of the history array
-        size_t _Ix;
-        for (_Ix = 0; _Ix < _Nx - _Mx; ++_Ix) { // fill in lower region
-            _Ty _Tmp       = (this->_Ax[_Ix + _Nx] & _HMSK) | (this->_Ax[_Ix + _Nx + 1] & _LMSK);
-            this->_Ax[_Ix] = (_Tmp >> 1) ^ (_Tmp & 1 ? _Px : 0) ^ this->_Ax[_Ix + _Nx + _Mx];
-        }
+        if constexpr (_M_mod_n == 0) {
+            for (size_t _Ix = 0; _Ix < _Nx - _One_mod_n; ++_Ix) { // fill in values
+                const _Ty _Tmp = (this->_Ax[_Ix + _Nx] & _HMSK) | (this->_Ax[_Ix + _Nx + _One_mod_n] & _LMSK);
+                this->_Ax[_Ix] = (_Tmp >> 1) ^ (_Tmp & 1 ? _Px : 0) ^ this->_Ax[_Ix + _Nx + _M_mod_n];
+            }
+
+            if constexpr (_One_mod_n == 1) { // fill in _Ax[_Nx - 1]
+                constexpr size_t _Ix = _Nx - _One_mod_n;

```

Comment:
This is the end condition of the above loop, so you could pull that up and name is something differently like `constexpr size_t _Last_ix = _Nx - _One_mod_n;`